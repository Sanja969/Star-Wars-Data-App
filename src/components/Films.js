import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import FilmsList from './FilmsList';

const Films = () => {
  const films = useSelector((state) => state.films);
  const characters = useSelector((state) => state.characters);
  const planets = useSelector((state) => state.planets);
  const ships = useSelector((state) => state.ships);
  const navigate = useNavigate();
  const [searchValue, setSearchValue] = useState('');

  return (
    <div className="text-white text-center pt-2">
      <div className="headline py-3">
        <h4>Star Wars Status</h4>
        <div className="d-flex flex-wrap justify-content-around">
          <div>
            <p className="h1 m-0">{films.length}</p>
            films
          </div>
          <div>
            <p className="h1 m-0">{characters.length}</p>
            characters
          </div>
          <div>
            <p className="h1 m-0">{planets.length}</p>
            planets
          </div>
          <div>
            <p className="h1 m-0">{ships.length}</p>
            ships
          </div>
        </div>
      </div>
      <div className="d-flex flex-column align-items-center justify-content-between p-3 categories">
        <h4>Choose category:</h4>
        <div className="d-flex justify-content-around w-100">
          <button className="list-unstyled h5 btn btn-outline-info" type="button" onClick={() => navigate('/planets')}>Planets</button>
          <button className="list-unstyled h5 btn btn-outline-info" type="button" onClick={() => navigate('/characters')}>Characters</button>
          <button className="list-unstyled h5 btn btn-outline-info" type="button" onClick={() => navigate('/starships')}>Ships</button>
        </div>
      </div>
      <input
        type="search"
        className="search w-75 text-center"
        placeholder="Search..."
        onChange={(event) => setSearchValue(event.target.value)}
      />
      <div className="w-100">
        <FilmsList films={films} searchValue={searchValue} />
      </div>

    </div>

  );
};

export default Films;
