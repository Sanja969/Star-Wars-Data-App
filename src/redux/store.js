import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import charactersReducer from './characters';
import planetsReducer from './planets';
import filmsReducer from './films';
import shipsReducer from './starships';
import detailReducer from './detail';

const rootReducer = combineReducers({
  characters: charactersReducer,
  planets: planetsReducer,
  films: filmsReducer,
  ships: shipsReducer,
  detail: detailReducer,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: [thunk, logger],
});

export default store;
