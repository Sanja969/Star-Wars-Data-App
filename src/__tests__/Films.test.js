import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Films from '../components/Films';

describe('Test Films render', () => {
  test('Test Snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <Films />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
