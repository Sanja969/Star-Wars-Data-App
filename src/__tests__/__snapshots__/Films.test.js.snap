// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Films render Test Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-white text-center pt-2"
      >
        <div
          class="headline py-3"
        >
          <h4>
            Star Wars Status
          </h4>
          <div
            class="d-flex flex-wrap justify-content-around"
          >
            <div>
              <p
                class="h1 m-0"
              >
                0
              </p>
              films
            </div>
            <div>
              <p
                class="h1 m-0"
              >
                0
              </p>
              characters
            </div>
            <div>
              <p
                class="h1 m-0"
              >
                0
              </p>
              planets
            </div>
            <div>
              <p
                class="h1 m-0"
              >
                0
              </p>
              ships
            </div>
          </div>
        </div>
        <div
          class="d-flex flex-column align-items-center justify-content-between p-3 categories"
        >
          <h4>
            Choose category:
          </h4>
          <div
            class="d-flex justify-content-around w-100"
          >
            <button
              class="list-unstyled h5 btn btn-outline-info"
              type="button"
            >
              Planets
            </button>
            <button
              class="list-unstyled h5 btn btn-outline-info"
              type="button"
            >
              Characters
            </button>
            <button
              class="list-unstyled h5 btn btn-outline-info"
              type="button"
            >
              Ships
            </button>
          </div>
        </div>
        <input
          class="search w-75 text-center"
          placeholder="Search..."
          type="search"
        />
        <div
          class="w-100"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-white text-center pt-2"
    >
      <div
        class="headline py-3"
      >
        <h4>
          Star Wars Status
        </h4>
        <div
          class="d-flex flex-wrap justify-content-around"
        >
          <div>
            <p
              class="h1 m-0"
            >
              0
            </p>
            films
          </div>
          <div>
            <p
              class="h1 m-0"
            >
              0
            </p>
            characters
          </div>
          <div>
            <p
              class="h1 m-0"
            >
              0
            </p>
            planets
          </div>
          <div>
            <p
              class="h1 m-0"
            >
              0
            </p>
            ships
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column align-items-center justify-content-between p-3 categories"
      >
        <h4>
          Choose category:
        </h4>
        <div
          class="d-flex justify-content-around w-100"
        >
          <button
            class="list-unstyled h5 btn btn-outline-info"
            type="button"
          >
            Planets
          </button>
          <button
            class="list-unstyled h5 btn btn-outline-info"
            type="button"
          >
            Characters
          </button>
          <button
            class="list-unstyled h5 btn btn-outline-info"
            type="button"
          >
            Ships
          </button>
        </div>
      </div>
      <input
        class="search w-75 text-center"
        placeholder="Search..."
        type="search"
      />
      <div
        class="w-100"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
